dd-sdk-android/src/main/kotlin/com/datadog/android/sessionreplay/internal/domain/RequestBodyFactory.kt

```diff
@@ -0,0 +1,111 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.domain
+
+import com.datadog.android.core.internal.utils.sdkLogger
+import com.datadog.android.log.Logger
+import com.datadog.android.sessionreplay.model.MobileSegment
+import com.datadog.android.sessionreplay.net.BytesCompressor
+import com.google.gson.JsonObject
+import okhttp3.MediaType
+import okhttp3.MultipartBody
+import okhttp3.RequestBody
+
+internal class RequestBodyFactory(
+    private val compressor: BytesCompressor = BytesCompressor(),
+    internal val internalLogger: Logger = sdkLogger
+) {
+
+    @Suppress("TooGenericExceptionCaught")
+    fun create(
+        segment: MobileSegment,
+        serializedSegment: JsonObject
+    ): RequestBody? {
+        return try {
+            // we need to add a new line at the end of each segment for being able to format it
+            // as an Array when read by the player
+            val segmentAsBinary = (serializedSegment.toString() + "\n").toByteArray()
+            buildRequestBody(segment, segmentAsBinary)
+        } catch (e: Throwable) {
```

Comment:
minor: we can limit it to only `e: Exception` (`Error` is unlikely can be recoverable or even be thrown by the code below) and also move `@Suppress("TooGenericExceptionCaught")` annotation directly to the `catch` block (`catch (@Suppress("TooGenericExceptionCaught" e: Exception)`)